deps/compile.jl

```diff
@@ -0,0 +1,182 @@
+using CUDAdrv
+using Compat
+
+"Database of compute capabilities with matching shader model, and initial version of the CUDA toolkit supporting this architecture."
+const architectures = [
+    # cap       SM          CUDA
+    # NOTE: CUDA versions only checked starting with v4.0
+    (v"1.0",    "sm_10",    v"4.0"),
+    (v"1.1",    "sm_11",    v"4.0"),
+    (v"1.2",    "sm_12",    v"4.0"),
+    (v"1.3",    "sm_13",    v"4.0"),
+    (v"2.0",    "sm_20",    v"4.0"),
+    (v"2.1",    "sm_21",    v"4.0"),
+    (v"3.0",    "sm_30",    v"4.2"),
+    (v"3.2",    "sm_32",    v"6.0"),
+    (v"3.5",    "sm_35",    v"5.0"),
+    (v"3.7",    "sm_37",    v"6.5"),
+    (v"5.0",    "sm_50",    v"6.0"),
+    (v"5.2",    "sm_52",    v"7.0"),
+    (v"5.3",    "sm_53",    v"7.5"),
+    (v"6.0",    "sm_60",    v"8.0"),
+    (v"6.1",    "sm_61",    v"8.0"),
+    (v"6.2",    "sm_62",    v"8.0") ]
+
+"Return the most recent supported architecture for a CUDA device."
+function architecture(dev::CuDevice; cuda=nothing)
+    cap = capability(dev)
+
+    # Devices are compatible with code generated for lower compute models
+    compat_architectures = filter(x -> x[1] <= cap, architectures)
+
+    if cuda !== nothing
+        compat_architectures = filter(x -> cuda >= x[3], compat_architectures)
+    end
+
+    if length(compat_architectures) == 0
+        error("No support for requested device or software (compute model <= $cap" *
+              cuda === nothing ? "" : ", CUDA >= $cuda).")
+    end
+
+    return compat_architectures[length(compat_architectures)][2]
+end
+
+type Toolchain
+    version::VersionNumber
+    nvcc::String
+    flags::Vector{String}
+end
+
+const toolchain = Ref{Toolchain}()
+
+function discover_toolchain()
+    # Check availability NVCC
+    if haskey(ENV, "NVCC")
+        nvcc = ENV["NVCC"]
+    elseif haskey(ENV, "CUDA_PATH")
```

Comment:
Many systems that I know use `CUDA_HOME` so it might be worth checking that as well and if `CUDA_PATH` is standard then we should use it as well for `find_libraries`